sample/rules.go

```diff
@@ -233,13 +234,16 @@ func extractValueFromSpan(trace *types.Trace, span *types.Span, condition *confi
 	}
 
 	// whether this condition is matched by this span.
-	var value any
-	var exists bool
 	for _, field := range condition.Fields {
 		// check if rule uses root span context
 		if strings.HasPrefix(field, RootPrefix) {
-			field = field[len(RootPrefix):]
-			span = trace.RootSpan
+			// make sure root span exists
+			if trace.RootSpan != nil {
+				field = field[len(RootPrefix):]
+				span = trace.RootSpan
+			} else {
+				continue
```

Comment:
```suggestion
				// we wanted root span but this trace doesn't have one, so just skip it
				continue
```

Reply:
Surely we should write the comment as "When a nil root comes along, you must skip it. Skip it good." 😀

Reply:
OK, I would approve a PR with that comment in a heartbeat. Wearing a lampshade and a shiny jacket.